### PR TITLE
Add Omid versions 1.1.1 & 1.1.2

### PR DESCRIPTION
**Pull request by soenkeliebau**
# Description

Adds Omid version 1.1.1 & 1.1.2.
Adds upload script for omid src archives.

**Changelog**

### BUG FIXES:

| JIRA | Summary | Priority | Component |
|:---- |:---- | :--- |:---- |
| [OMID-280](https://issues.apache.org/jira/browse/OMID-280) | Use Hbase 2.5 for building OMID |  Critical | . |
| [OMID-278](https://issues.apache.org/jira/browse/OMID-278) | Change default waitStrategy to LOW\_CPU |  Major | . |
| [OMID-266](https://issues.apache.org/jira/browse/OMID-266) | Remove and ban unrelocated Guava from Omid |  Major | . |
| [OMID-277](https://issues.apache.org/jira/browse/OMID-277) | Omid 1.1.2 fails with Phoenix 5.2 |  Major | . |

### IMPROVEMENTS:

| JIRA | Summary | Priority | Component |
|:---- |:---- | :--- |:---- |
| [OMID-275](https://issues.apache.org/jira/browse/OMID-275) | Expose backing HBase Table from TTable |  Major | . |
| [OMID-249](https://issues.apache.org/jira/browse/OMID-249) | Improve default network address logic |  Major | . |
| [OMID-272](https://issues.apache.org/jira/browse/OMID-272) | Support JDK17 |  Major | . |
| [OMID-258](https://issues.apache.org/jira/browse/OMID-258) | Bump maven plugins/dependencies to latest |  Major | . |
| [OMID-264](https://issues.apache.org/jira/browse/OMID-264) | Fix deprecated WARNING in check-license stage |  Minor | . |

### BUG FIXES:

| JIRA | Summary | Priority | Component |
|:---- |:---- | :--- |:---- |
| [OMID-250](https://issues.apache.org/jira/browse/OMID-250) | Remove duplicate declarations of hadoop-hdfs-client dependency in pom.xml |  Trivial | . |
| [OMID-240](https://issues.apache.org/jira/browse/OMID-240) | Transactional visibility is broken |  Critical | . |
| [OMID-248](https://issues.apache.org/jira/browse/OMID-248) | Transactional Phoenix tests fail on Java 17  in getDefaultNetworkInterface |  Major | . |
| [OMID-237](https://issues.apache.org/jira/browse/OMID-237) | TestHBaseTransactionClient.testReadCommitTimestampFromCommitTable fails |  Major | . |
| [OMID-247](https://issues.apache.org/jira/browse/OMID-247) | Change TSO default port to be outside the ephemeral range |  Critical | . |
| [OMID-246](https://issues.apache.org/jira/browse/OMID-246) | Update Surefire plugin to 3.0.0 and switch to TCP forkNode implementation |  Major | . |
| [OMID-236](https://issues.apache.org/jira/browse/OMID-236) | Upgrade Netty to 4.1.86.Final |  Major | . |

### SUB-TASKS:

| JIRA | Summary | Priority | Component |
|:---- |:---- | :--- |:---- |
| [OMID-254](https://issues.apache.org/jira/browse/OMID-254) | Upgrade to phoenix-thirdparty 2.1.0 |  Major | . |
| [OMID-256](https://issues.apache.org/jira/browse/OMID-256) | Bump hbase and other dependencies to latest version |  Major | . |
| [OMID-253](https://issues.apache.org/jira/browse/OMID-253) | Upgrade Netty to 4.1.100.Final |  Major | . |
| [OMID-255](https://issues.apache.org/jira/browse/OMID-255) | Upgrade guava to 32.1.3-jre |  Major | . |
| [OMID-257](https://issues.apache.org/jira/browse/OMID-257) | Upgrade bouncycastle and move from jdk15on to latest jdk18on |  Major | . |

### OTHER:

| JIRA | Summary | Priority | Component |
|:---- |:---- | :--- |:---- |
| [OMID-251](https://issues.apache.org/jira/browse/OMID-251) | Bump license-maven-plugin to latest version |  Major | . |
| [OMID-245](https://issues.apache.org/jira/browse/OMID-245) | Add dependency management for Guava to use 32.1.1 |  Major | . |
| [OMID-244](https://issues.apache.org/jira/browse/OMID-244) | Upgrade SnakeYaml version to 2.0 |  Major | . |
| [OMID-242](https://issues.apache.org/jira/browse/OMID-242) | Bump guice version to 5.1.0 to support JDK 17 |  Major | . |
| [OMID-241](https://issues.apache.org/jira/browse/OMID-241) | Add logging to TSO server crash |  Major | . |
| [OMID-239](https://issues.apache.org/jira/browse/OMID-239) | OMID TLS support |  Major | . |
| [OMID-234](https://issues.apache.org/jira/browse/OMID-234) | Bump SnakeYaml version to 1.33 |  Major | . |

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Does your change affect an SBOM? Make sure to update all SBOMs
```

### CODE REVIEW

**Comment by lfrancke** [comment]
Can this cause problems again: https://github.com/apache/phoenix-omid/pull/154/files#diff-6c4b5f133b92f9439af086250e44f2d31e3c8f67c949b2f0f9f06d40dcd8858cR144-R161

I don't have any evidence that it will but I'd feel better if someone else looks at it again.
Want to avoid another forward/reverse lookup DNS confusion issue.

**Comment by soenkeliebau** [comment]
It might cause issues ... not sure tbh.

Not because of a reverse lookup snafu or tls stuff I think though. That code as far as I can tell determines the address that tso thinks it should be reachable at, and that address is then written into ZK for Omid clients to use when disovering their TSO server.
If this is resolved to something other cannot reach .. 

I've put this down for now .. will try and test it over the weekend.

File: omid/Dockerfile [hunk 1]
```diff
@@ -14,10 +14,11 @@ USER stackable
 WORKDIR /stackable
 
 # hadolint ignore=DL3003
-RUN --mount=type=cache,id=maven,uid=1000,target=/stackable/.m2/repository <<EOF
-  curl --fail -L https://dist.apache.org/repos/dist/release/phoenix/phoenix-omid-${PRODUCT}/phoenix-omid-${PRODUCT}-src.tar.gz | tar -xzC .
+RUN --mount=type=cache,id=omid${PRODUCT},uid=1000,target=/stackable/.m2/repository <<EOF
```
Comment (razvan) [inline omid/Dockerfile:17]:
```suggestion
RUN --mount=type=cache,id=maven-omid-${PRODUCT},uid=1000,target=/stackable/.m2/repository <<EOF
```

**Review by razvan — APPROVED** [review]
lgtm! one suggestion

**Review by razvan — APPROVED** [review]

**Pull request merged**